monkey/monkey_island/cc/agent_event_handlers/scan_event_handler.py

```diff
@@ -85,6 +97,21 @@ def _update_nodes(self, target_machine: Machine, event: ScanEvent):
             src_machine.id, target_machine.id, CommunicationType.SCANNED
         )
 
+    def _update_tcp_connections(self, src_node: Node, target_machine: Machine, event: TCPScanEvent):
+        node_connections = dict(deepcopy(src_node.tcp_connections))
+        try:
+            machine_connections = set(node_connections[target_machine.id])
+        except KeyError:
+            machine_connections = set()
+        open_ports = [port for port, status in event.ports.items() if status == PortStatus.OPEN]
+        for open_port in open_ports:
```

Comment:
Since `open_ports` is only used once, prefer generator comprehension to list comprehension. Even for small lists, it's something like 3x faster.

Reply:
I take it back. I did some more testing. In this case, list comprehension is better.